// capacitor-notifications-listener/dist/plugin.cjs.js
'use strict';

var core = require('@capacitor/core');

const NotificationsListener = core.registerPlugin('NotificationsListener', {
    web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.NotificationsListenerWeb()),
});

// browser can't read your notifications.
class NotificationsListenerWeb extends core.WebPlugin {
    constructor() {
        super(...arguments);
        this.errorMessage = 'Method will never be implemented. Whose notifications you want to read?';
    }
    attachListener() {
        throw new Error(this.errorMessage);
    }
    startListening() {
        throw new Error(this.errorMessage);
    }
    restoreCachedNotifications() {
        throw new Error(this.errorMessage);
    }
    stopListening() {
        throw new Error(this.errorMessage);
    }
    requestPermission() {
        throw new Error(this.errorMessage);
    }
    isListening() {
        throw new Error(this.errorMessage);
    }
    replacePackagesWhitelist() {
        throw new Error(this.errorMessage);
    }
}

var web = /*#__PURE__*/Object.freeze({
    __proto__: null,
    NotificationsListenerWeb: NotificationsListenerWeb
});

exports.NotificationsListener = NotificationsListener;
